### PR TITLE
Add redis storage

### PR DESCRIPTION
**Pull request by eyurtsev**
Add a redis implementation of a BaseStore

### CODE REVIEW

File: libs/langchain/langchain/storage/redis.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            )
+
+        self.client = client
+
+        if not isinstance(ttl, int) and ttl is not None:
+            raise TypeError(f"Expected int or None, got {type(ttl)} instead.")
+
+        self.ttl = ttl
+        self.namespace = namespace
+        self.namespace_delimiter = "/"
```
Comment (baskaryan) [inline libs/langchain/langchain/storage/redis.py:67]: worth making this configurable? if not, does it need to be stored as attribute (allows folks to mess with it)

File: libs/langchain/langchain/storage/redis.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            # Delete keys
+            redis_store.mdelete(["key1"])
+
+            # Iterate over keys
+            for key in redis_store.yield_keys():
+                print(key)
+    """
+
+    def __init__(
+        self, client: Any, *, ttl: Optional[int] = None, namespace: Optional[str] = None
```
Comment (baskaryan) [inline libs/langchain/langchain/storage/redis.py:37]: could type hint client if you wanted to

**Review by baskaryan — APPROVED** [review]
2 optional nits, lgtm!

**Comment by eyurtsev** [comment]
Merging and following up with PR to address remaining issues (since they were not blocking)

**Pull request merged**